platform/macos/godot_main_macos.mm

```diff
@@ -51,15 +50,43 @@ int main(int argc, char **argv) {
 	setrlimit(RLIMIT_STACK, &stack_lim);
 #endif
 
-	int first_arg = 1;
-	const char *dbg_arg = "-NSDocumentRevisionsDebugMode";
+	LocalVector<char *> args;
+	args.resize(argc);
+	uint32_t argsc = 0;
+
+	int wait_for_debugger = 0; // wait 5 second by default
+
 	for (int i = 0; i < argc; i++) {
-		if (strcmp(dbg_arg, argv[i]) == 0) {
-			first_arg = i + 2;
+		if (strcmp("-NSDocumentRevisionsDebugMode", argv[i]) == 0) {
+			// remove "-NSDocumentRevisionsDebugMode" and the next argument
+			i++;
+			continue;
+		}
+
+#if TOOLS_ENABLED
+		if (strcmp("--os-debug", argv[i]) == 0) {
+			i++;
+			wait_for_debugger = 5000; // wait 5 seconds by default
+			if (i < argc && strncmp(argv[i], "--", 2) != 0) {
+				wait_for_debugger = atoi(argv[i]);
+				i++;
+			}
+			continue;
```

Comment:
For macOS, we can added a `--os-debug <ms>` run argument, so you can have it wait for a debugger and attach to debug the entire startup process when the Godot editor launches the child process, which was very handy for this work (and many other things I've done in the past)

Reply:
Probably should be in `ifdef DEBUG_ENABLED` block.